torch/_inductor/utils.py

```diff
@@ -429,35 +431,103 @@ def aggregate_origins(node_schedule):
         return set()
 
 
-def get_fused_kernel_name(node_schedule, descriptive_names):
+def get_module_name_from_meta(meta_dict):
+    mod_name = ""
+    torch_op = ""
+    if "nn_module_stack" in meta_dict:
+        # nn_module_stack is set in multiple places and can be organized
+        # differently.  The key is always the module name.  The value
+        # can either be a tuple or a string.  When it is a tuple, the
+        # last element of the tuple contains the torch_op.  When it is
+        # a string, the string is the torch_op.
+        # Example contents of nn_module_stack
+        # {'L__self___conv1': ("L['self'].conv1",
+        #                 <class 'torch.nn.modules.conv.Conv2d'>)}
+        # {'L__self___bn3': ("L['self'].bn3",
+        #                 <class 'torch.nn.modules.batchnorm.BatchNorm2d'>)}
+        # The key is the unique module name.  The value contains the
+        # torch_op type eg. <class 'torch.nn.modules.batchnorm.BatchNorm2d'>
+        mod_name, extra = list(meta_dict["nn_module_stack"].items())[-1]
```

Comment:
cc @SherlockNoMad @ydwu4 do you know if there are any nn_module_stack utilities that would replace this ?